src/librustc_lint/builtin.rs

```diff
@@ -1918,3 +1918,65 @@ impl EarlyLintPass for Async2018 {
         }
     }
 }
+
+
+pub struct ExplicitOutlivesRequirements;
+
+impl LintPass for ExplicitOutlivesRequirements {
+    fn get_lints(&self) -> LintArray {
+        lint_array![EXPLICIT_OUTLIVES_REQUIREMENTS]
+    }
+}
+
+impl ExplicitOutlivesRequirements {
+    fn fire_lint(&self, cx: &LateContext, lifetime: &hir::Lifetime, span: Span) {
+        let is_static = match lifetime.name {
+            hir::LifetimeName::Static => true,
+            _ => false
+        };
+        if is_static && !cx.tcx.features().infer_static_outlives_requirements {
```

Comment:
Should the `is_static` condition be dropped here as well? Ideally we wouldn't fire lints unless the fix works becaue the crate already has the feature enabled

Reply:
`infer_static_outlives_requirements` (note, not the same as `infer_outlives_requirements`) is a [new feature gate](https://github.com/rust-lang/rust/issues/44493#issuecomment-407848932) specifically for the special case of static lifetimes that there [were concerns](https://github.com/rust-lang/rust/issues/44493#issuecomment-407846046) about.

I was reluctant to gate the entire lint on `infer_outlives_requirements` since we're so close to stabilizing it, but I guess it's not very much work (and not much work to undo when we stabilize, even if that's only a few weeks from now) ...

Reply:
Ah yeah due to the way `cargo fix` works it's best to ignore this lint unless the feature is turned on as otherwise lots of false positives get emitted